src/town_cmd.cpp

```diff
@@ -1565,8 +1563,10 @@ static void DoCreateTown(Town *t, TileIndex tile, uint32 townnameparts, TownSize
 	UpdateTownRadius(t);
 	t->flags = 0;
 	t->cache.population = 0;
-	t->grow_counter = 0;
-	t->growth_rate = 250;
+	/* Spread growth across ticks so even if there are many
+	 * similar towns they're unlikely to grow all at once */
+	t->grow_counter = ((uint32)_tick_counter + t->index) % TOWN_GROWTH_TICKS;
```

Comment:
Again, I would drop "_tick_counter" here.
It rather confuses than affects stuff.

Reply:
I actually don't rly like it from competitive CB perspective, but it makes sense since if you create a big map with a lot of towns many of them will have exact same growth_rate so will be growing in same tick which I assume can even be a bit laggy in some cases (haven't tested though). I was thinking of making default growth rate calculation to be smooth rather than jump in 50-house steps then it will probably be ok to remove this part but before that I'd leave it be.

Reply:
@frosch123 or did you mean drop _tick_counter but leave t->index? Yeah, that would probably work.

Reply:
Yes, leave t->index for the spreading of CPU resources.
But remove _tick_counter since it does not do anything.